sql/plan/ddl_trigger.go

```diff
@@ -0,0 +1,91 @@
+// Copyright 2020 Liquidata, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package plan
+
+import (
+	"fmt"
+	"github.com/liquidata-inc/go-mysql-server/sql"
+)
+
+type TriggerOrder struct {
+	PrecedesOrFollows string // PrecedesStr, FollowsStr
+	OtherTriggerName string
+}
+
+type CreateTrigger struct {
+	triggerName string
+	triggerTime string
+	triggerEvent string
+	triggerOrder *TriggerOrder
+	table sql.Node
+	body sql.Node
+	bodyString string
+}
+
+func NewCreateTrigger(triggerName, triggerTime, triggerEvent string, triggerOrder *TriggerOrder, table sql.Node, body sql.Node, bodyString string) *CreateTrigger {
+	return &CreateTrigger{
+		triggerName:  triggerName,
+		triggerTime:  triggerTime,
```

Comment:
Thinking on it more, this goes for Time and Event too. Let's make these "golang enums" as well, and verify their values in the `parse` package.

Reply:
The vitess parser already will verify them at parse time (only the strings BEFORE, AFTER, etc will parse), but agree on the need for validation on other code paths. Will work on this for the public trigger interface.